src/transformStateWithClones.js

```diff
@@ -1,13 +1,42 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object} stateCopy
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultsOfActions = [];
+  let stateCopy = { ...state };
+
+  for (const objectFromActionsArray of actions) {
+    switch (objectFromActionsArray.type) {
+      case 'addProperties':
+
+        Object.assign(stateCopy, objectFromActionsArray.extraData);
```

Comment:
You can destructure all the needed props of action (type, extraData, keysToRemove)